clcache.py

```diff
@@ -737,6 +737,21 @@ def expandCommandLine(cmdline):
 class CommandLineAnalyzer(object):
 
     @staticmethod
+    def _preprocessToStdout(options):
+        assert ('P' in options or 'E' in options or 'EP' in options), "must be in preprocessor mode"
+
+        # Note: For MSVS 2013 and 2015 there is a documentation bug in
+        # https://msdn.microsoft.com/en-us/library/becb7sys.aspx stating that
+        # "To send the preprocessed output to stdout, with #line directives,
+        # use /P and /EP together."
+        # The actual compiler behavior is that both /P and /P /EP go to file.
+        # /P is with #line annotations and /P /EP is without.
+        if 'P' in options:
+            return False
+        else:
+            return True
```

Comment:
I agree on the value of assertions in general, but for a method call `_preprocessToStdout` I think it's perfectly fine if it yields `True` if the given options indicate that preprocessed output will be printed to `stdout` and `False` otherwise. I.e. I think it's plausible that this method returns `False` for e.g. empty lists.